plan/plan.go

```diff
@@ -98,12 +98,16 @@ type Plan interface {
 	GetID() string
 	// Check whether this plan is correlated or not.
 	IsCorrelated() bool
+	// Set the value of correlated.
```

Comment:
This method needs more comment.
I checked all the implementations, the rule seems like below

1. if any child plan is correlated this plan is correlated, except for Apply.
2. Apply is correlated if first child is correlated, or second child has unknown correlated column.
3, If any expression in this plan is correlated, the plan is correlated.

If this rule is correct, we can add it in comment to make it easy to read.

Another thing to comment is when this function is called.
It seem like for any plan, this method is called as the final step of building when all the children and expressions have been set.